tests/client/core/actions/test_index.js

```diff
@@ -28,6 +32,28 @@ describe('core actions setClientApp', () => {
   });
 });
 
+describe('core actions setUserAgent', () => {
+  it('creates the SET_USER_AGENT action', () => {
+    const userAgent = userAgents.chrome[0];
+    const { browser, os } = UAParser(userAgent);
+
+    assert.deepEqual(actions.setUserAgent(userAgent), {
+      type: 'SET_USER_AGENT',
+      payload: { userAgent, userAgentInfo: { browser, os } },
+    });
+  });
+
+  it('allow an empty userAgent value', () => {
+    const userAgent = '';
+    const { browser, os } = UAParser(userAgent);
+
+    assert.deepEqual(actions.setUserAgent(''), {
+      type: 'SET_USER_AGENT',
+      payload: { userAgent, userAgentInfo: { browser, os } },
+    });
+  });
```

Comment:
I do have a test for empty strings and a test that uses `'tofubrowser'` which is not valid. But it's in `App.js` and aims to test something else so I'll add one here as well.